File: frontend/lib/restaurant/restaurant.ts
import type { ItemAPI } from "../item/item";

export type RestaurantAPI = {
    readonly items: readonly ItemAPI[];
    readonly restaurant: {
        readonly address: string;
        readonly cep: string;
        readonly city: string;
        readonly district: string;
        readonly email: string;
        readonly name: string;
        readonly state: string;
        readonly telephone: string;
    };
};

File: backend/src/restaurant.rs
struct User {
    cnpj: String,
    name: String,
    telephone: String,
    cep: String,
    address: String,
    district: String,
    city: String,
    state: String,
    email: String,
    hash: String,
    salt: String,
    created_at: String,
    updated_at: String,
}

struct CreateUserDTO {
    cnpj: String,
    name: String,
    telephone: String,
    cep: String,
    address: String,
    district: String,
    city: String,
    state: String,
    email: String,
    password: String,
}

struct LoginUserDTO {
    email: String,
    password: String,
}

struct UpdateUserDTO {
    cnpj: String,
    name: String,
    telephone: String,
    cep: String,
    address: String,
    district: String,
    city: String,
    state: String,
    email: String,
    password: String,
}
